sympy/polys/matrices/ddm.py

```diff
@@ -92,6 +92,25 @@ def __init__(self, rowslist, shape, domain):
         if not (len(self) == m and all(len(row) == n for row in self)):
             raise DDMBadInputError("Inconsistent row-list/shape")
 
+    def getitem(self, i, j):
+        return self[i][j]
+
+    def extract_slice(self, slice1, slice2):
+        from .domainmatrix import getindex
+        m, n = self.shape
+        i, _m = getindex(slice1, m)
+        j, _n = getindex(slice2, n)
+
+        if i >= m:
+            raise IndexError("row index out of range")
+        if j >= n:
+            raise IndexError("column index out of range")
```

Comment:
Slicing doesn't normally raise.

Reply:
For the same reason as above, these things are handled by `DomainMatrix.__getitem__` for `SDM.extract_slice`.